testing/examples/WORKSPACE

```diff
@@ -43,20 +43,23 @@ _java_image_repos()
 
 container_pull(
```

Comment:
@gregmagolan this looks like a chance for us to document that surprising behavior that the tag is ignored if a digest is present, do I have that right?

Reply:
It's documented as part of the `tag` attribute for [`container_pull`](https://github.com/bazelbuild/rules_docker/blob/master/docs/container.md#container_pull).

![Screenshot 2021-11-22 at 10 12 34](https://user-images.githubusercontent.com/9749173/142843108-5b134118-32b8-4a58-bc1d-8d49606f17da.png)

Reply:
@alexeagle Just want to confirm [the above comment](https://github.com/bazelbuild/rules_docker/pull/1964/files#r754339183) is for me?

Reply:
yes please, I'd rather not introduce digests into the example unless we warn users about that

Reply:
I believe @alexeagle is referring to commenting in the BUILD file as the current behaviour is suprising:

```
    name = "alpine_linux_amd64",
    digest = "sha256:954b378c375d852eb3c63ab88978f640b4348b01c1b3456a024a81536dafbbf4",
    registry = "index.docker.io",
    repository = "library/alpine",
    # tag field is ignored since digest is set
    tag = "3.8",
```

I also think it should be an error if you specify both since it is too easy to update the tag and think that your change is load bearing

Reply:
I was imagining that we should error when you provide both `digest` and `tag` since the latter is ignored, but @gravypod points out that bazel's `git_repository` does the same. I guess we could warn but it's not actionable. We could document and recommend the separate dictionary scheme that @gregmagolan did for a client. WDYT?

To merge this PR I'm happy with any comment on the `tag` field so users are aware.